detox/android/detox/src/full/java/com/wix/detox/espresso/matcher/ViewMatchers.kt

```diff
@@ -54,6 +64,42 @@ fun toHaveSliderPosition(expectedValuePct: Double, tolerance: Double): Matcher<V
         }
     }
 
+fun withRegexText(jsRegex: String): Matcher<View> =
+    object: BoundedMatcher<View, TextView>(TextView::class.java) {
+        override fun describeTo(description: Description) {
+            description.appendText("withRegexText($jsRegex)")
+        }
+
+        override fun matchesSafely(view: TextView): Boolean {
+            val text = view.text?.toString()
+            return text?.isMatch(jsRegex) ?: false
+        }
+    }
+
+fun withRegexTagValue(jsRegex: String): Matcher<View> =
+    object: BoundedMatcher<View, View>(View::class.java) {
+        override fun describeTo(description: Description) {
+            description.appendText("withRegexTagValue($jsRegex)")
+        }
+
+        override fun matchesSafely(view: View): Boolean {
+            val tag = view.tag?.toString()
+            return tag?.isMatch(jsRegex) ?: false
+        }
+    }
+
+fun withRegexContentDescription(jsRegex: String): Matcher<View> =
+    object: BoundedMatcher<View, View>(View::class.java) {
+        override fun describeTo(description: Description) {
+            description.appendText("withRegexContentDescription($jsRegex)")
+        }
+
+        override fun matchesSafely(view: View): Boolean {
+            val contentDescription = view.contentDescription?.toString()
+            return contentDescription?.isMatch(jsRegex) ?: false
+        }
+    }
```

Comment:
You can avoid this duplicated structure by adding a factory method for those regex-matchers:

```
private fun createRegexMatcher(matcherDescription: String, regex: String, extractStringFromView: (View) -> String?): Matcher<View> =
    object: BoundedMatcher<View, View>(View::class.java) {
        override fun describeTo(description: Description) {
            description.appendText("$matcherDescription($regex)")
        }

        override fun matchesSafely(view: View): Boolean {
            val extractedString = extractStringFromView(view)
            return extractedString?.isMatch(regex) ?: false
        }
    }
```

It might even be worth extracting it to another file 🤔

Reply:
Usage example:

```kotlin
private fun withRegexText(jsRegex: String): Matcher<View> =
    createRegexMatcher("withRegexText", jsRegex) { view -> (view as? TextView)?.text?.toString() }
```

Reply:
Also, those methods should be private since you're using them in this file context only

Reply:
Another small note, sorry if it seems nitpicky, but personally, I prefer ordering methods according to their "caller-callee" relationship, where a method follows its caller when possible (if the method has not been declared before). This approach improves readability and mirrors the logical flow of execution in the code

For example, in this order, `withRegexText` will follow `withText`